test/modules/core/lib/pick-layers.spec.js

```diff
@@ -0,0 +1,424 @@
+        // {
+        //   funcName: 'pickMultipleObjects',
+        //   positions: {
+        //     x: 260,
+        //     y: 300
+        //   },
+        //   count: 1
+        // },
+        {
+          funcName: 'pickMultipleObjects',
+          positions: {
+            x: 10,
+            y: 10
+          },
+          count: 0
+        }
+      ]
+    }
+  },
+  {
+    id: 'multiLayers',
+    props: {
+      width: 500,
+      height: 550,
+      layers: [
+        new ScatterplotLayer({
+          data: DATA.points,
+          getPosition: d => d.COORDINATES,
+          getColor: [255, 128, 0],
+          getRadius: d => d.SPACES,
+          opacity: 0.1,
+          pickable: true,
+          radiusScale: 30,
+          radiusMinPixels: 1,
+          radiusMaxPixels: 30
+        }),
+        new PolygonLayer({
+          data: DATA.polygons,
+          getPolygon: f => f,
+          getFillColor: () => [255 * Math.random(), 0, 0],
+          getLineColor: [0, 0, 0, 255],
+          getLineDashArray: [20, 0],
+          getWidth: 20,
+          getElevation: () => Math.random() * 1000,
+          opacity: 0.1,
+          pickable: true,
+          lineDashJustified: true,
+          elevationScale: 0.6
+        }),
+        new PathLayer({
+          data: DATA.zigzag,
+          opacity: 0.6,
+          getPath: f => f.path,
+          getColor: [128, 0, 0],
+          getWidth: 10,
+          getDashArray: [20, 0],
+          widthMinPixels: 1,
+          pickable: true
+        })
+      ],
+      views: [new MapView()],
+      viewState: VIEW_STATE,
+      useDevicePixels: false
+    },
+    pickingMethods: {
+      singlePixel: [
+        {
+          funcName: 'pickObject',
+          positions: {
+            x: 260,
+            y: 300
+          },
+          count: 1
+        },
+        {
+          funcName: 'pickObject',
+          positions: {
+            x: 10,
+            y: 10
+          },
+          count: 0
+        }
+      ],
+      rectangle: [
+        {
+          funcName: 'pickObjects',
+          positions: {
+            x: 0,
+            y: 0,
+            width: 400,
+            height: 400
+          },
+          count: 32
+        },
+        {
+          funcName: 'pickObjects',
+          positions: {
+            x: 10,
+            y: 10,
+            width: 10,
+            height: 10
+          },
+          count: 0
+        }
+      ],
+      multiDepth: [
+        // {
+        //   funcName: 'pickMultipleObjects',
+        //   positions: {
+        //     x: 260,
+        //     y: 300
+        //   },
+        //   count: 1
+        // },
+        {
+          funcName: 'pickMultipleObjects',
+          positions: {
+            x: 10,
+            y: 10
+          },
+          count: 0
+        }
+      ]
+    }
+  }
+];
+
+for (const testCase of TEST_CASES) {
+  // eslint-disable-next-line no-loop-func
+  test(`picking#${testCase.id}`, t => {
+    const deck = new Deck();
```

Comment:
Could there be a race condition here (we reset the `onAfterRender` callback before it executes, or it executes twice)?

For a separate PR: We should have a way to 1) start deck.gl without animationLoop and 2) to force a deck.gl render so that we can run this synchronously, rather than using the `onAfterRender` callback. I seem to recall that the changes made to support mapbox integration offers such facilities, but perhaps as undocumented methods.

Suggest opening an issue to track this.